src/nvim/os/shell.c

```diff
@@ -119,14 +119,14 @@ int os_call_shell(char_u *cmd, ShellOpts opts, char_u *extra_args)
   }
 
   size_t nread;
-  int status = shell((const char *)cmd,
-                     (const char *)extra_args,
-                     input.data,
-                     input.len,
-                     output_ptr,
-                     &nread,
-                     emsg_silent,
-                     forward_output);
+
+  int status = do_os_system(shell_build_argv((char *)cmd, (char *)extra_args),
```

Comment:
Not really related to this PR, but if I haven't missed something it looks like `extra_args` is only ever used in `mch_expand_wildcards`. Perhaps this function could be refactored  to manipulate the argv list directly, instead of complicating the entire `call_shell` `os_call_shell` `shell_build_argv` chain. (later, no need to do it in this PR).